virtual_ecosystem/models/abiotic/energy_balance.py

```diff
@@ -247,562 +236,335 @@ def calculate_longwave_emission(
+    specific_heat_capacity_soil: float | NDArray[np.float32],
+    time_interval: int,
+) -> NDArray[np.float32]:
+    """Update soil temperature using heat diffusion.
 
-    conductivity_from_soil = (
-        topsoil_moisture * soil_saturated_vapour_pressure
-    ).to_numpy()
-
-    # Factors from leaf and air temperature linearisation
-    a_A, b_A = leaf_and_air_temperature_linearisation(
-        conductivity_from_ref_height=(
-            current_conductivities["conductivity_from_ref_height"][
-                layer_structure.index_filled_canopy
-            ]
-        ),
-        conductivity_from_soil=conductivity_from_soil,
-        leaf_air_heat_conductivity=(
-            current_conductivities["leaf_air_heat_conductivity"][
-                layer_structure.index_filled_canopy
-            ]
-        ),
-        air_temperature_ref=air_temperature_ref.to_numpy(),
-        top_soil_temperature=topsoil_temperature.to_numpy(),
-    )
+    TODO add description and equations
 
-    # Factors from longwave radiative flux linearisation
-    a_R, b_R = longwave_radiation_flux_linearisation(
-        a_A=a_A,
-        b_A=b_A,
-        air_temperature_ref=air_temperature_ref.to_numpy(),
-        leaf_emissivity=abiotic_constants.leaf_emissivity,
-        stefan_boltzmann_constant=core_constants.stefan_boltzmann_constant,
-    )
+    Args:
+        ground_heat_flux: Ground heat flux at top soil, [W m-2]
+        soil_temperature: Soil temperature for each soil layer, [C]
+        soil_thermal_conductivity: Thermal conductivity of soil [W m-2 K-1]
+        soil_bulk_density: Soil bulk density, [kg m-3]
+        specific_heat_capacity_soil: Specific heat capacity of soil [J kg-1 K-1]
+        soil_layer_thickness: Thickness of each soil layer, [m]
+        time_interval: Time interval, [s]
 
-    # Factors from vapour pressure linearisation
-    delta_v_ref = calculate_slope_of_saturated_pressure_curve(
-        air_temperature_ref.to_numpy(),
-        saturated_pressure_slope_parameters=(
-            abiotic_constants.saturated_pressure_slope_parameters
-        ),
-    )
+    Returns:
+        Updated soil temperatures, [C]
+    """
 
-    a_E, b_E = vapour_pressure_linearisation(
-        vapour_pressure_ref=vapour_pressure_ref.to_numpy(),
-        saturated_vapour_pressure_ref=saturated_vapour_pressure_ref.to_numpy(),
-        soil_vapour_pressure=soil_vapour_pressure.to_numpy(),
-        conductivity_from_soil=conductivity_from_soil,
-        leaf_vapour_conductivity=(
-            current_conductivities["leaf_vapour_conductivity"][
-                layer_structure.index_filled_canopy
-            ]
-        ),
-        conductivity_from_ref_height=(
-            current_conductivities["conductivity_from_ref_height"][
-                layer_structure.index_filled_canopy
-            ]
-        ),
-        delta_v_ref=delta_v_ref,
-    )
+    n_layers = len(soil_temperature)
+    new_soil_temperature = soil_temperature.copy()
```

Comment:
Why do you need to use `.copy()` here?

Reply:
Yeah - from the code below, it looks like `new_soil_temperature = np.full_like(soil_temperature, np.nan)` would be more explicit?

Reply:
quite possibly a relict, will adjust